feat(cart): 尝试添加 cart 服务

suyiiyii opened the pull request.

suyiiyii committed:
feat(cart): 重构配置模块并引入 viper

- 使用 viper 替代 ioutil 和 yaml 处理配置文件
- 移除 GetEnv 函数，简化配置流程
- 更新 go.mod 文件，添加新的依赖项
- 删除 conf 文件夹下的环境特定配置文件

suyiiyii committed:
refactor(cart): 重构日志输出逻辑

- 移除了复杂的日志配置代码
- 将日志输出改为使用 os.Stdout- 删除了不必要的导入语句

suyiiyii committed:
feat(cart): 添加环境变量支持并更新数据库连接方式

- 引入 godotenv 包以支持 .env 文件加载
- 更新 MySQL 配置，将 DSN拆分为单独的主机、端口、用户名和密码字段
- 在配置初始化中添加环境变量自动读取和绑定
- 修改数据库连接方式，使用新的配置字段构建 DSN

suyiiyii committed:
refactor(cart): 初始化购物车响应结构

- 在 GetCartService.Run 方法中添加了日志输出- 初始化了 GetCartResp 结构，设置了用户 ID 和空的购物车项列表

suyiiyii committed:
chore: gen dal code

suyiiyii committed:
feat(cart): 添加购物车商品逻辑

- 实现了添加商品到购物车的业务逻辑
- 新增了根据用户 ID 和商品 ID 查询购物车商品的方法
- 更新了购物车商品模型，将字段类型改为 uint32
-移除了 Redis 初始化代码
- 添加了 MySQL 初始化和查询默认设置

suyiiyii committed:
feat(cart): 实现空购物车和获取购物车功能

- 在 cart_items 表中添加 deleted_at 字段，用于软删除
- 实现 EmptyCartService 来清空用户购物车
- 实现 GetCartService 来获取用户购物车商品信息

Tsukilc commented on app/cart/biz/service/get_cart.go:
这里不return resp吗，别的地方好像也有

suyiiyii replied:
把方法丢给ai问问为什么return后面可以不加东西？（

suyiiyii committed:
build(app/cart): 添加 Dockerfile 和 .dockerignore 文件

- 新增 Dockerfile 用于构建和运行应用程序
- 新增 .dockerignore 文件以排除不必要的文件和目录
- Dockerfile 包含两个阶段：构建阶段和最终运行阶段
- 构建阶段使用 golang:1.23.4镜像，下载依赖并编译应用程序
- 最终运行阶段使用 debian:12-slim镜像，创建非特权用户并复制可执行文件
-暴露 8080端口并设置启动命令

suyiiyii committed:
build(cart): 更新数据库相关依赖- 添加 gorm.io/plugin/dbresolver v1.5.3 依赖
- 更新 gorm.io/driver/mysql 到 v1.5.7
- 更新 gorm.io/gorm 到 v1.25.12
- 更新 gorm.io/driver/sqlite 到 v1.5.0
- 更新 gorm.io/driver/sqlserver到 v1.5.4- 更新 gorm.io/driver/postgres 到 v1.5.0

suyiiyii committed:
feat(cart): 嵌入配置文件到可执行文件

- 使用 go:embed 指令将 conf.yaml 文件嵌入到可执行文件中
- 用 bytes包装 configFile 代替 filepath 和 ioutil 读取配置文件
-移除了 confFileRelPath 变量和相关的路径操作

PengJingzhao reviewed (commented):
可以
